internal/cloud/realm/secrets_ext_test.go

```diff
@@ -0,0 +1,60 @@
+package realm_test
+
+import (
+	"testing"
+
+	"github.com/10gen/realm-cli/internal/app"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	u "github.com/10gen/realm-cli/internal/utils/test"
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+func TestSecrets(t *testing.T) {
+	u.SkipUnlessRealmServerRunning(t)
+
+	t.Run("should fail without an auth client", func(t *testing.T) {
+		client := realm.NewClient(u.RealmServerURL() + "")
+
+		_, err := client.Secrets(primitive.NewObjectID().Hex(), primitive.NewObjectID().Hex())
+		assert.Equal(t, realm.ErrInvalidSession{}, err)
+	})
+
+	t.Run("with an active session ", func(t *testing.T) {
+		client := newAuthClient(t)
+		groupID := u.CloudGroupID()
+
+		secretName := "secretName"
+		secretValue := "secretValue"
+
+		testApp, appErr := client.CreateApp(groupID, "secrets-test", realm.AppMeta{})
+		assert.Nil(t, appErr)
+		defer client.DeleteApp(groupID, testApp.ID)
```

Comment:
[q] I added a defer here but if we want to be more explicit here, especially if we wanted to log any error that could come up, I can refactor this. I felt this was fine since there isn't much we can do here, the impact would be a floating app that we can easily attribute to this test, and we have e2e tests already covering app delete

Reply:
yeah, honestly I haven't been a good citizen here in regards to test cleanup yet, so let's make a helper here in the `realm_test` package, something like:

```go
func setupTestApp(t *testing.T, groupID, name string) (realm.App, func()) {
  t.Helper()
  // create app with empty meta here
  // assert create err was nil
  // teardown := func() { /* call delete app, you can use `t.Log` to print back any errors */ }
}
```

Reply:
yeah, honestly I haven't been a good citizen here in regards to test cleanup yet, so let's make a helper here in the `realm_test` package, something like:

```go
func setupTestApp(t *testing.T, groupID, name string) (realm.App, func()) {
  t.Helper()
  // create app with empty meta here
  // assert create err was nil
  // teardown := func() { /* call delete app, you can use `t.Log` to print back any errors */ }
}
```

Reply:
Done! Added a helpers_test.go file. I also added the realm.Client as an argument

Reply:
instead of `helpers_test.go`, just place it in the existing `app_test.go` file

Reply:
Done! I left app_test as is (i.e. didn't use the helper there). I am fine with modifying app_test if we want to reuse the helper inside of there (I was a bit split on that). I also don't mind updating the other tests in a follow-up/this PR